src/main/java/htsjdk/samtools/SamInputResource.java

```diff
@@ -147,6 +164,9 @@ public String toString() {
             case FILE:
                 childToString = asFile().toString();
                 break;
+            case PATH:
+                childToString = asPath().toString();
```

Comment:
This is an existing issue with all the other resource types, so I've left it to be consistent. This should really be fixed separately.